src/libraries/Microsoft.Extensions.Hosting/src/HostingHostBuilderExtensions.cs

```diff
@@ -134,6 +134,18 @@ public static IHostBuilder ConfigureContainer<TContainerBuilder>(this IHostBuild
             return hostBuilder.ConfigureContainer<TContainerBuilder>((context, builder) => configureDelegate(builder));
         }
 
+        /// <summary>
+        /// Adds a delegate for configuring the host settings. This can be called multiple times and
+        /// the results will be additive.
+        /// </summary>
+        /// <param name="hostBuilder">The <see cref="IHostBuilder" /> to configure.</param>
+        /// <param name="configure">The delegate for configuring the <see cref="HostOptions"/>.</param>
+        /// <returns>The same instance of the <see cref="IHostBuilder"/> for chaining.</returns>
+        public static IHostBuilder ConfigureHostOptions(this IHostBuilder hostBuilder, Action<HostOptions> configure)
```

Comment:
How does this work?

Reply:
This implementation is not working, I fixed it. But the tests for some reason do not pass. Implementation via .ConfigureServices and pass configureDelegate to collection.Configure